src/libraries/System.Runtime/tests/System.Globalization.Tests/Invariant/InvariantMode.cs

```diff
@@ -559,6 +559,7 @@ public static IEnumerable<object[]> GetUnicode_TestData()
         }
 
         [ConditionalFact(nameof(PredefinedCulturesOnlyIsDisabled))]
+        [SkipOnPlatform(TestPlatforms.iOS | TestPlatforms.MacCatalyst | TestPlatforms.tvOS, "We are using the system icu and not loading it directly on these platforms")]
```

Comment:
This does not look right. These tests are testing InvariantGlobalization. Why are we initializing the ICU support with InvariantGlobalization? 

The flavor of the ICU should not matter.

Reply:
I think this bug in the placeholder shims introduced in this change. The bug should be fixed there, and not worked around here.

Reply:
The reason why I think skipping this test makes sense is the for iOS, we no longer have to explicitly load ICU at all. iOS does that on its own regardless.

Reply:
I think the problem is that this method https://github.com/dotnet/runtime/pull/96684/files#diff-30efd6ad50033567c6a4d713b941bdb23461aec9cc3ceba02ed29db716151b2aR168-R171 should not be unconditionally returning 1. It makes us think that there is successfully loaded ICU and its version is 1, even where there is no ICU loaded at all (not even the OS provided one).

Reply:
What do you think would be the proper way? Return 0 out of `GlobalizationNative_GetICUVersion`?

Reply:
I think it would best if the real native ICU shims are never called for hybrid globalization and their implementation  asserts, similar to: https://github.com/dotnet/runtime/blob/cc5f1df48e5e39a6fa5ab94ca4664aac3cc36898/src/native/libs/System.Native/pal_datetime.c#L90-L93

Looking at the code, I am not sure what is the expected behavior of  `PlatformDetection.IsIcuGlobalization` for hybrid globalization. Is `PlatformDetection.IsIcuGlobalization` supposed to true or false for hybrid globalization?

The code does not have consistent assumptions about `PlatformDetection.IsIcuGlobalization` for hybrid globalization. For example:
 https://github.com/dotnet/runtime/blob/cc5f1df48e5e39a6fa5ab94ca4664aac3cc36898/src/libraries/System.Runtime/tests/System.Globalization.Tests/CultureInfo/GetCultureInfo.cs#L36 implies that `PlatformDetection.IsIcuGlobalization`  should be false for hybrid globalization. If it was not the case, the condition can be simplified to just `(PlatformDetection.IsIcuGlobalization`.

On the other hand, https://github.com/dotnet/runtime/blob/cc5f1df48e5e39a6fa5ab94ca4664aac3cc36898/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs#L390 implies that `PlatformDetection.IsIcuGlobalization` should be true for hybrid globalization. If it was not the case, the `IsIcuGlobalizationAndNotHybrid` property is not needed.

Reply:
(Also, to make things less confusing - `IsHybridGlobalizationOnOSX` should be renamed to `IsHybridGlobalizationOnApplePlatform`. It includes all Apple OSes, not just OSX.)

Reply:
All makes sense, thanks. I would prefer to do the property rename in a follow up.

Reply:
Hybrid mode initially was a combination of loading ICU libs and using native functions, that is why there was a need for https://github.com/dotnet/runtime/blob/cc5f1df48e5e39a6fa5ab94ca4664aac3cc36898/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs#L391.
`HybridGlobalization` is also used for Browser and for Browser we still load ICU and use native functions.
Now for mobile Apple we don't load at all ICU and we don't call to native ICU shims but because of this call from tests https://github.com/dotnet/runtime/blob/fdb03ca8776f5197fb046067b994b605bb6e715c/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs#L424 it calls `GlobalizationNative_GetICUVersion` and test was failing.

For renaming `IsHybridGlobalizationOnOSX`  there is already ticket https://github.com/dotnet/runtime/issues/96328

Reply:
If the `IsIcuGlobalizationAndNotHybrid` property is no longer needed, could you please add it to the cleanup list (https://github.com/dotnet/runtime/issues/96328)?

Reply:
Added, thanks.